newrelic-agent/src/main/java/com/newrelic/agent/jmx/values/EmbeddedTomcatJmxValues.java

```diff
@@ -0,0 +1,111 @@
+    private static final JmxMetric CONNECTIONS_IDLE = DataSourceJmxMetricGenerator.CONNECTIONS_IDLE.createMetric("NumIdle");
+    private static final JmxMetric CONNECTIONS_MAX = DataSourceJmxMetricGenerator.CONNECTIONS_MAX.createMetric("MaxActive");
+    private static final JmxMetric CONNECTIONS_MAX_TOMCAT_8 = DataSourceJmxMetricGenerator.CONNECTIONS_MAX.createMetric("MaxTotal");
+
+    // THREAD POOL METRICS
+    private static final JmxMetric CURRENT_MAX_COUNT = ServerJmxMetricGenerator.MAX_THREAD_POOL_COUNT.createMetric("maxThreads");
+    private static final JmxMetric CURRENT_ACTIVE_COUNT = ServerJmxMetricGenerator.ACTIVE_THREAD_POOL_COUNT.createMetric("currentThreadsBusy");
+    private static final JmxMetric CURRENT_IDLE_COUNT = JmxMetric.create(new String[] { "currentThreadCount",
+            "currentThreadsBusy" }, MetricNames.JMX_THREAD_POOL_IDLE, JmxAction.SUBTRACT_ALL_FROM_FIRST, JmxType.SIMPLE);
+
+    private final List<BaseJmxValue> metrics = new ArrayList<>(METRIC_COUNT);
+
+    public EmbeddedTomcatJmxValues() {
+        createMetrics("*");
+    }
+
+    public EmbeddedTomcatJmxValues(String name) {
+        createMetrics(name);
+
+    }
+
+    private void createMetrics(String name) {
+        /*
+         * Only used by 7.0. The manager bean provides information about sessions. sessionCounter is the total number of
+         * sessions created by this manager. ActiveSessions is the number of active sessions at this moment.
+         * expiredSesions is the number of sessions that have expired. RejectedSessions is the number of sessions
+         * rejected due to maxActive being reached. SessionAverageAliveTime is the average time an expired session had
+         * been alive.
+         */
+        metrics.add(new BaseJmxValue(name + ":type=Manager,context=*,host=*,*", MetricNames.JMX_SESSION + "{context}/",
+                new JmxMetric[] { ACTIVE_SESSIONS, EXPIRED_SESSIONS, REJECTED_SESSIONS, SESSION_ALIVE_TIME }));
+        /* This is for 6.0 and 5.5. */
+        metrics.add(new BaseJmxValue(name + ":type=Manager,path=*,host=*", MetricNames.JMX_SESSION + "{path}/",
+                new JmxMetric[] { ACTIVE_SESSIONS, EXPIRED_SESSIONS, REJECTED_SESSIONS, SESSION_ALIVE_TIME }));
+        /*
+         * Provides information about the thread pool. The current thread count and the current number of threads which
+         * are busy.
+         */
+        metrics.add(new BaseJmxValue(name + ":type=ThreadPool,name=*", MetricNames.JMX_THREAD_POOL + "{name}/",
+                new JmxMetric[] { CURRENT_ACTIVE_COUNT, CURRENT_IDLE_COUNT, CURRENT_MAX_COUNT }));
+
+        /*
+         * Provides information about the data source by finding the number of active and idle connections and the max connections.
+         * In Tomcat 7, the number of max connections is represented by the maxActive attribute. In tomcat 8
```

Comment:
Comment should be updated to reflect whatever the new mbean ends up being (and why it is different)